File: src/main/java/de/hse/model/ProductCategory.java
package de.hse.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "product_category")
public class ProductCategory {
    @Id
    @GeneratedValue(strategy= GenerationType.SEQUENCE)
    @Column(name = "product_category_id")
    private long productCategoryId;

    @Column(name = "product_id")
    private long productId;

    @Column(name = "category_id")
    private long categoryId;

//    @ManyToOne
//    private Product product;
//
//    @ManyToOne
//    private Category category;
}

File: src/web/src/app/model/product.category.ts
import { Category } from './category';
import { Product } from './product';

export class ProductCategory {
  productCategoryId?: number;
  categoryId?: number;
  productId?: number;
  product?: Product;
  category?: Category;

  constructor(
      productCategoryId?: number,
      categoryId?: number,
      productId?: number,
      product?: Product,
      category?: Category
    ) {
        this.productCategoryId = productCategoryId;
        this.productId = productId;
        this.categoryId = categoryId;
        this.product = product;
        this.category = category;
      }
}
